### PR TITLE
Fix: claim colony funds after mint tokens motion is finalized

### PR DESCRIPTION
**Pull request by mmioana**
## Description

- Trigger the MOTION_CLAIM_MINT_TOKENS saga after clicking the Finalize button for a Mint tokens action
- Note: Created a new saga as we might want to re-work how we're claiming incoming funds - currently the block ingestor will claim all available funds within a colony for the given token

## Testing

TODO: Incoming funds resulted from a MINT_TOKENS motion are claimed and no longer appear as funds that need to be accepted. 
Screenshot 2024-08-13 at 12 37 00

* Step 1. Enable `Multi-sig Permissions` extension (no need to change the default settings)
* Step 2. Give `leela` Owner permissions to Take actions via Multi-sig
Screenshot 2024-08-13 at 12 46 13
* Step 3. Create a Mint Tokens with Multi-Sig decision method
Screenshot 2024-08-13 at 12 46 49
* Step 4. Finalize motion
Screenshot 2024-08-13 at 12 47 15
* Step 5. Check minted tokens are already claimed in the Incoming Funds page
Screenshot 2024-08-13 at 12 47 32
* Step 6. Enable `Reputation Weighted (Lazy Consensus method)` extension 
* Step 7. Choose `Testing governance` as extension settings and enable it
Screenshot 2024-08-13 at 12 48 47
* Step 8. Create Mint Tokens with Reputation decision method
* Step 9. Support motion by staking at least 10%
Screenshot 2024-08-13 at 12 50 10
* Step 10. Run `npm run forward-time 10`
* Step 11. Refresh page and finalize motion
Screenshot 2024-08-13 at 12 51 04
* Step 12. Check minted tokens are already claimed in the Incoming Funds page
Screenshot 2024-08-13 at 12 51 18

## Diffs

**New stuff** ✨

* MOTION_CLAIM_MINT_TOKENS redux saga

Resolves [#2913](https://github.com/JoinColony/colonyCDapp/issues/2913)

### CODE REVIEW

**Review by rdig — CHANGES_REQUESTED** [review]
Very nice work! I've tested this and seems to be working as expected. Both Multisig and Motions will auto-claim the incoming funds

Screenshot from 2024-08-14 01-27-05
Screenshot from 2024-08-14 01-27-38
Screenshot from 2024-08-14 01-27-45
Screenshot from 2024-08-14 01-31-46
Screenshot from 2024-08-14 01-31-55

However I noticed a issue _(which might or might not be related to this)_. The colony balance doesn't update. No matter if it's via Multisig or via Motions, after a mint and automatic claim, the balance of the colony stays the same

Screenshot from 2024-08-14 01-29-37
Screenshot from 2024-08-14 01-29-49

This needs to be fixed, preferably here, as it's major problem if the balance gets out of date with the chain state.

**Comment by mmioana** [comment]
Thank you @rdig for reviewing it 🙏 I'll check the issue regarding the balance update

**Comment by mmioana** [comment]
Hey @rdig I tried to mint a larger amount of tokens and the balance seems to be updating

https://github.com/user-attachments/assets/03480b2e-cd6a-4c11-ad0a-d2af04651b3b


I'm wondering if the balance update was not visible only because there was a too small amount of tokens minted? 🤔

File: src/components/v5/common/ActionSidebar/partials/hooks.ts [hunk 1]
```diff
@@ -81,3 +88,44 @@ export const useIsFieldDisabled = () => {
 
   return false;
 };
+
+export const useFinalizeSuccessCallback = (): FinalizeSuccessCallback => {
+  const { startPollingColonyData, stopPollingColonyData } = useColonyContext();
```
Comment (bassgeta) [inline src/components/v5/common/ActionSidebar/partials/hooks.ts:93]: Dropped you a DM also, but I think we can do this without polling and it should work just fine :ok_hand:

**Review by bassgeta — CHANGES_REQUESTED** [review]
Works like a charm, maybe we can remove polling so it's a bit less API spammy :D  
Other than that, balance before:
image
Finalized a mint tokens Multi-sig
image
And it's updated :sunglasses: 
image

**Review by jakubcolony — CHANGES_REQUESTED** [review]
Nice start on the PR @mmioana!

I noticed this issue is overlapping with #2506. Ideally, the solution for both multi sig and "regular" motions would be to include the claim transaction in the motion's action. However, as far as I'm aware, it is not supported by multisig (and definitely not supported by regular motions). This is great as it gets us unblocked in both cases.

The only reason for requesting changes is the saga duplication, we've always had the ability to claim funds (e.g. the "Accept all funds" button in the Incoming Funds page) via the `CLAIM_TOKEN` redux action and it should be used here as well.

**Comment by mmioana** [comment]
> Nice start on the PR @mmioana!
> 
> I noticed this issue is overlapping with #2506. Ideally, the solution for both multi sig and "regular" motions would be to include the claim transaction in the motion's action. However, as far as I'm aware, it is not supported by multisig (and definitely not supported by regular motions). This is great as it gets us unblocked in both cases.
> 
> The only reason for requesting changes is the saga duplication, we've always had the ability to claim funds (e.g. the "Accept all funds" button in the Incoming Funds page) via the `CLAIM_TOKEN` redux action and it should be used here as well.

Hey @jakubcolony thank you for noticing the overlapping issue 🌟 

The reason I decided to create the new saga was due to the fact we might want to re-work how we're claiming the incoming funds, as currently we're accepting all incoming funds and not only the ones related to a specific motion/action. 

However, this actually needs to be clarified with Product (will do so) and maybe then we can create a separate issue 🙌

**Review by bassgeta — APPROVED** [review]
Nice solution, just a handler and stuff **just** works :D
Automatic claim of my 111 CREDS multisig
image
image

Voting reputation:
image
image

**Comment by mmioana** [comment]
@jakubcolony after discussing with Arren indeed it seems there is no issue in claiming all incoming funds for the same token in a colony. So I'll proceed with removing the duplicated saga 🙌

**Review by jakubcolony — APPROVED** [review]
> @jakubcolony after discussing with Arren indeed it seems there is no issue in claiming all incoming funds for the same token in a colony. So I'll proceed with removing the duplicated saga 🙌

Sounds good! It is actually impossible to "choose" funds you claim, you have to do it for the entire balance in given token.

Thanks for addressing my feedback, all looks good now 👍 

Before finalizing the motion:
image

image


After:
image

**Comment by mmioana** [comment]
> > @jakubcolony after discussing with Arren indeed it seems there is no issue in claiming all incoming funds for the same token in a colony. So I'll proceed with removing the duplicated saga 🙌
> 
> Sounds good! It is actually impossible to "choose" funds you claim, you have to do it for the entire balance in given token.
> 
> Thanks for addressing my feedback, all looks good now 👍
> 
> Before finalizing the motion: image
> 
> image
> After: image

Thank you @jakubcolony for all the valuable feedback! 🌟

**Pull request merged**